isolate.go

```diff
@@ -83,17 +83,17 @@ func (i *Isolate) IsExecutionTerminating() bool {
 // no compile-time options but can be provided to eagerly compile or
 // consume from code cache.
 // error will be of type `JSError` if not nil.
-func (i *Isolate) CompileScript(source, origin string, option ScriptCompilerCompileOption) (ScriptCompilerCachedData, error) {
+func (i *Isolate) CompileUnboundScript(source, origin string, option ScriptCompilerCompileOption) (*UnboundScript, error) {
```

Comment:
I think we can just name it `CompileScript`, since it is on Isolate rather than on Context

```suggestion
func (i *Isolate) CompileScript(source, origin string, option ScriptCompilerCompileOption) (*UnboundScript, error) {
```

Reply:
I think this might be confusing since it returns a type of unbound script and `script` is already an actual object in the v8 api

Reply:
I was thinking that we would just add Context.CompileScript once we have that other type.  However, maybe you are right that it could still be confusing, especially considering that V8's API has a CompileUnboundScript function, although on the v8::ScriptCompiler namespace